### DISCUSSION TITLE
auth.tesla Error

### DISCUSSION THREAD

**Discussion by galarfar** [teslamate-org/teslamate#1368] [General]
I get this message in the LOG when I try to log in to Teslamate, are there any who experience the same?

GET https://auth.tesla.com/oauth2/v3/authorize -> error: "timeout"

Have tried a lot but nothing seems to solve the problem :( Restart docker and download latest update for teslamate/grafana, nothing helps.

**Comment by SorenLan** [comment]
Same issue here. Timeout when talking to Tesla for about 15 hours...

`teslamate_1  | 2021-02-20 13:05:04.381 [info] Refreshing access token ...

teslamate_1  | 2021-02-20 13:06:04.497 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "timeout" (60115.382 ms)

teslamate_1  | 2021-02-20 13:06:04.497 [error] GenServer TeslaMate.Api terminating

teslamate_1  | ** (MatchError) no match of right hand side value: {:error, %TeslaApi.Error{env: nil, message: "timeout", reason: :token_refresh}}

teslamate_1  |     (teslamate 1.21.4) lib/teslamate/api.ex:151: TeslaMate.Api.handle_info/2

teslamate_1  |     (stdlib 3.14) gen_server.erl:689: :gen_server.try_dispatch/4

teslamate_1  |     (stdlib 3.14) gen_server.erl:765: :gen_server.handle_msg/6

teslamate_1  |     (stdlib 3.14) proc_lib.erl:226: :proc_lib.init_p_do_apply/3

teslamate_1  | Last message: :refresh_auth

teslamate_1  | 2021-02-20 13:06:12.631 [info] GET /

teslamate_1  | 2021-02-20 13:06:12.633 [info] Sent 302 in 1ms

teslamate_1  | 2021-02-20 13:07:04.558 [error] POST https://auth.tesla.com/oauth2/v3/token -> error: "timeout" (60058.213 ms)

teslamate_1  | 2021-02-20 13:07:04.558 [warn] Token refresh failed: %TeslaApi.Error{env: nil, message: "timeout", reason: :token_refresh}

teslamate_1  | 2021-02-20 13:07:04.558 [info] Scheduling token refresh in 1 h`

**Reply by mrfleetwood** [reply]
I've similar errors too. Problems with authentication and constant Health check failed notifications too.

**Comment by meavydev** [comment]
I'm guessing this is related to https://github.com/adriankumpf/teslamate/issues/1369
I have only just turned on 2FA, but it's working fine from Android.
